src/tracing/tests/TestMetricEvents.cpp

```diff
@@ -0,0 +1,354 @@
+/*
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <lib/support/UnitTestRegistration.h>
+#include <tracing/backend.h>
+#include <tracing/metric_event.h>
+
+#include <nlunit-test.h>
+
+#include <algorithm>
+#include <string>
+#include <vector>
+
+using namespace chip;
+using namespace chip::Tracing;
+
+namespace chip {
+namespace Tracing {
+inline std::string str(MetricKey key)
+{
+    return std::string(key);
+}
+
+inline bool operator==(const chip::Tracing::MetricEvent & lhs, const chip::Tracing::MetricEvent & rhs)
```

Comment:
- Inline is not required.
- Fully qualified `chip::Tracing` is not required since the code is already withing the chip::Tracing namespace.